service/history/workflow/cache/cache.go

```diff
@@ -171,6 +180,29 @@ func (c *CacheImpl) GetOrCreateWorkflowExecution(
 	return weCtx, weReleaseFunc, err
 }
 
+func (c *CacheImpl) tryRestore(
+	ctx context.Context,
+	shardCtx shard.Context,
+	wfCtx *workflow.ContextImpl,
+) error {
+	if wfCtx.GetWorkflowKey().RunID == "" {
+		return nil
+	}
+	if r, ok := c.shardCtx.(interface{ DoWorkflowContextRestoration() bool }); ok && !r.DoWorkflowContextRestoration() {
```

Comment:
I dislike this but there are dozens and dozens of unit tests that have been written to assume that loading a workflow.Context from this cache does not require a NamespaceRegistry or a functioning Mutable state, both of which are required if we're going restore parts of context (namely the UpdateRegistry) from MutableState. Additionally, many unit tests are _also_ coupled to the accidental truth that CacheImpl produces the concrete type *workflow.ContextImpl, so injecting a more clever workflow.Context implementation (e.g. one that always returns an empty set of incomplete update records) also does not work because it breaks all the downcasting.